sys_siglist deprecated

<!--To help us understand and resolve your issue, please fill out the form to the best of your ability. Feel free to delete the sections that do not apply.-->

**Description of the problem**

Compiling GMT 6.1.0
OS: Arch Linux 5.4.59-1-lts
Compiler: gcc 10.2.0-1
CLib: glibc 2.32-2

`sys_siglist` seems to have been removed from the GNU C Library 2.32. It's no longer declared in signal.h and I guess it has been removed from [string.h too](https://sourceware.org/pipermail/libc-announce/2020/000029.html) (beginning of August).

By redeclaring it in signal.h, I can complile past gmt_common_sighandler.c.o (in version gmt-6.1.0), but it fails at the very end with 
```
[ 96%] Linking C shared module plugins/supplements.so
[100%] Built target supplements
Scanning dependencies of target gmt
[100%] Building C object src/CMakeFiles/gmt.dir/gmt.c.o
[100%] Linking C executable gmt
/usr/bin/ld: libgmt.so.6.1.0: undefined reference to `sys_siglist'
collect2: error: ld returned 1 exit status
make[2]: *** [src/CMakeFiles/gmt.dir/build.make:121: src/gmt] Error 1
make[1]: *** [CMakeFiles/Makefile2:769: src/CMakeFiles/gmt.dir/all] Error 2
make: *** [Makefile:171: all] Error 2
```
so it seems it may just be gone from glibc 2.32 totally.

I don't code in C much and since I don't actively need to use GMT 6.1 at the moment, I'm not very motivated to dig further, so I'd thought I'd just bring it up here.

Let me know if you want more information.

--- a/cmake/modules/ConfigureChecks.cmake
+++ b/cmake/modules/ConfigureChecks.cmake
@@ -160,9 +160,10 @@ check_function_exists (strdup           HAVE_STRDUP)
 check_function_exists (strndup          HAVE_STRNDUP)
 check_function_exists (strsep           HAVE_STRSEP)
 check_function_exists (strtod           HAVE_STRTOD)
+check_function_exists (strtod           HAVE_STRTOD)
 # Note: trailing underscore = GDAL workaround
 check_function_exists (strtof           HAVE_STRTOF_)
-check_function_exists (strtok_r         HAVE_STRTOK_R)
+check_function_exists (strsignal        HAVE_STRSIGNAL)
 
 if (WIN32)
 	check_function_exists (_fseeki64      HAVE__FSEEKI64)

--- a/src/gmt_common_sighandler.c
+++ b/src/gmt_common_sighandler.c
@@ -194,7 +194,11 @@ void sig_handler(int sig_num, siginfo_t *info, void *ucontext) {
 		return;
 	}
 	else {
+#ifdef HAVE_STRSIGNAL
+		fprintf (stderr, "ERROR: Caught signal number %d (%s) at\n", sig_num, strsignal(sig_num);
+#else
 		fprintf (stderr, "ERROR: Caught signal number %d (%s) at\n", sig_num, sys_siglist[sig_num]);
+#endif
 		backtrace_symbols_fd (array, 2, STDERR_FILENO); /* print function with faulting instruction */
 		size = backtrace (array, 50); /* get void*'s for all entries on the stack */
 		fprintf (stderr, "Stack backtrace:\n");

